supportedBackends/htcondor/src/main/scala/cromwell/backend/impl/htcondor/HtCondorJobExecutionActor.scala

```diff
@@ -141,26 +182,39 @@ class HtCondorJobExecutionActor(override val jobDescriptor: BackendJobDescriptor
     }
   }
 
-  /**
-    * Abort a running job.
-    */
-  override def abort(): Unit = Future.failed(new UnsupportedOperationException("HtCondorBackend currently doesn't support aborting jobs."))
+  private def calculateHash(): String = {
+    val sb = StringBuilder.newBuilder
+    val cmd = call.task.instantiateCommand(jobDescriptor.inputs, callEngineFunction, identity) match {
+      case Success(cmd) => cmd
+      case Failure(ex) =>
+        val errMsg = s"$tag Cannot instantiate job command for caching purposes due to ${ex.getMessage}."
+        log.error(ex.getCause, errMsg)
+        throw new IllegalStateException(errMsg, ex.getCause)
+    }
+    sb.append(cmd)
+    sb.append(runtimeAttributes.failOnStderr)
+    sb.append(runtimeAttributes.dockerImage.getOrElse(""))
+    DigestUtils.md5Hex(sb.toString())
+  }
 
-  override def preStart(): Unit = {
-    log.debug("{} Creating execution folder: {}", tag, executionDir)
-    executionDir.toString.toFile.createIfNotExists(true)
+  private def createExecutionFolderAndScript(): Unit = {
     try {
```

Comment:
I know some people don't mind in mixing try catch with Try utility of Scala. But, here Scala Try may be helpful because you are calling this function inside receive of an actor and as throwing exception, IMO better could be you return Try[] from this function and then in failure case send message to yourself `self ! blah..` that will run your receive `case message: Any` to report failure response otherwise ignored in former implementation (Correct me if i am wrong ?).